src/main/java/com/linecorp/armeria/client/routing/NodeSelectStrategy.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+/**
+ * A factory interface of {@link NodeSelector}
+ */
+public interface NodeSelectStrategy<T extends Node> {
+    enum SelectStrategyType {
```

Comment:
Yeah, sort of. I realized that we don't need factory methods, though:

```java
public interface NodeSelectStrategy... {
    NodeSelectStrategy<Node> ROUND_ROBIN = new RoundRobinStrategy();
    NodeSelectStrategy<WeightedNode> WEIGHTED_ROUND_ROBIN = new Weighted...();
    ...
}
```

WDYT?